inc/ThirdParty/Plugins/CDN/Cloudflare.php

```diff
@@ -358,6 +360,25 @@ public function purge_cloudflare_partial( $urls ) {
 		$this->facade->purge_urls( $post_ids );
 	}
 
+	/**
+	 * Purges Cloudflare cache after automatic cache purge (cache lifespan)
+	 *
+	 * @since 3.20.2
+	 *
+	 * @param array $deleted Array of deleted cache data.
+	 * @param array $args    Array with lifespan and file_age_limit data.
+	 *
+	 * @return void
+	 */
+	public function purge_cloudflare_after_automatic_purge( $deleted, $args ) {
+		if ( ! $this->is_plugin_active() ) {
+			return;
+		}
+
+		// For cache lifespan purge, we do a full cache purge since it affects the entire site.
+		$this->facade->purge_everything();
```

Comment:
Should we be more precise and purge urls that were deleted, I believe we can get this from the deleted param in the hook. Maybe this is a product decision. @DahmaniAdame What do you think?

CC @wordpressfan